crates/aggchain-proof-builder/src/lib.rs

```diff
@@ -261,27 +261,80 @@ impl<ContractsClient> AggchainProofBuilder<ContractsClient> {
             .await
             .map_err(Error::UnableToFetchTrustedSequencerAddress)?;
 
-        // From the request
-        let inserted_gers: Vec<InsertedGER> = request
+        // Helper function to filter, sort, and convert a vector of items to the desired
+        // output type. It filters items based on a range of blocks, sorts them
+        // using type Ord implementation, and maps them to a new type using the
+        // provided mapping function.
+        fn filter_sort_map<T, F, U>(
```

Comment:
Nit: In order to make the method more generic, it could take `items` as `impl IntoIterator<Item = T>` and return an `impl Iterator<Item = U>`. That would avoid a number of intermediate `Vec`s introduced below.